tests/statemodels/portstate_maintenance_test.py

```diff
@@ -0,0 +1,129 @@
+        assert matching_portstate_pm.matches_event(portstate_event, state)
+
+    def test_should_return_false_for_events_that_are_not_portstate_events(
+        self, state, reachability_event, matching_portstate_pm
+    ):
+        assert not matching_portstate_pm.matches_event(reachability_event, state)
+
+    def test_should_return_false_for_non_matching_events(self, state, reachability_event, nonmatching_portstate_pm):
+        assert not nonmatching_portstate_pm.matches_event(reachability_event, state)
+
+
+class TestMatchesPortstate:
+    @pytest.mark.parametrize("portstate_pm", [MatchType.REGEXP, MatchType.STR, MatchType.INTF_REGEXP], indirect=True)
+    def test_should_return_false_for_non_matching_port(self, portstate_pm, device, port):
+        port.ifdescr = "wrongport"
+        assert not portstate_pm.matches_portstate(device, port)
+
+    @pytest.mark.parametrize("portstate_pm", [MatchType.REGEXP, MatchType.STR], indirect=True)
+    def test_regexp_and_str_should_return_true_for_matching_port(self, portstate_pm, device, port):
+        assert portstate_pm.matches_portstate(device, port)
+
+    @pytest.mark.parametrize("portstate_pm", [MatchType.INTF_REGEXP], indirect=True)
+    def test_intf_regexp_should_return_true_for_matching_port_and_device(self, portstate_pm, device, port):
+        assert portstate_pm.matches_portstate(device, port)
+
+    @pytest.mark.parametrize("portstate_pm", [MatchType.INTF_REGEXP], indirect=True)
+    def test_intf_regexp_match_type_should_return_false_for_matching_port_and_non_matching_device(
+        self, portstate_pm, device, port
+    ):
+        device.name = "wrongdevice"
+        assert not portstate_pm.matches_portstate(device, port)
+
+    @pytest.mark.parametrize("portstate_pm", [MatchType.EXACT], indirect=True)
+    def test_should_return_false_if_invalid_match_type(self, portstate_pm, device, port):
+        assert not portstate_pm.matches_portstate(device, port)
+
+
+@pytest.fixture
+def portstate_pm(request, device, port) -> PortStateMaintenance:
+    return PortStateMaintenance(
+        start_time=datetime.datetime.now() - datetime.timedelta(days=1),
+        end_time=datetime.datetime.now() + datetime.timedelta(days=1),
+        match_type=request.param,
+        match_expression=port.ifdescr,
+        match_device=device.name,
+    )
+
+
+@pytest.fixture
+def matching_portstate_pm(device, port) -> PortStateMaintenance:
+    with patch("zino.statemodels.PortStateMaintenance.matches_portstate") as mock:
+        mock.return_value = True
+        yield PortStateMaintenance(
+            start_time=datetime.datetime.now() - datetime.timedelta(days=1),
+            end_time=datetime.datetime.now() + datetime.timedelta(days=1),
+            match_type="str",
+            match_expression=port.ifdescr,
+            match_device=device.name,
+        )
+
+
+@pytest.fixture
+def nonmatching_portstate_pm(device, port) -> PortStateMaintenance:
```

Comment:
Sonarlint complains here for the type hint: `Remove this yield statement or annotate function "matching_device_pm" with "typing.Generator" or one of its supertypes.`